### PR TITLE
Syntactical fixes

### PR DESCRIPTION
**Pull request by naveen-ram**
applied autopep8 as well as pylint and covered most syntax issues in the project. Linting through GitHub Actions now gives no errors or warnings.

### CODE REVIEW

File: .github/workflows/pythonpackage.yml [hunk 1]
```diff
@@ -24,8 +24,12 @@ jobs:
         pip install poetry
         poetry config virtualenvs.create false
         poetry install
-    - name: Lint with flake8
+    - name: Lint with autopep8 and flake8
       run: |
+        pip install autopep8
+        autopep8 --in-place --aggressive --aggressive base.py
+        autopep8 --in-place --aggressive --aggressive backend.py
+        autopep8 --in-place --aggressive --aggressive requirements.py
```
Comment (kadler) [inline .github/workflows/pythonpackage.yml]: I'm not sure why we're running autopep8 here. Isn't it a code formatter not a linter?

File: sqlalchemy_ibmi/backend.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+""" This module specifies the specific dialect for the AS400 using the PyODBCConnector"""
```
Comment (kadler) [inline sqlalchemy_ibmi/backend.py]:
```suggestion
""" This module specifies the the IBM i dialect using the PyODBCConnector"""
```

Comment (kadler) [reply sqlalchemy_ibmi/backend.py]: We should not be adding any new comments/code/whatever that uses the term AS400.

File: sqlalchemy_ibmi/base.py [hunk 3]
```diff
@@ -19,124 +19,151 @@
...
-    'uescape', 'decimal', 'percent_rank', 'unknown', 'deref', 'percentile_cont',
-    'unnest', 'element', 'percentile_disc', 'upper', 'exec', 'power', 'var_pop', 'exp',
-    'real', 'var_samp', 'false', 'recursive', 'varchar', 'filter', 'ref', 'varying',
-    'float', 'regr_avgx', 'width_bucket', 'floor', 'regr_avgy', 'window', 'fusion',
-    'regr_count', 'within', 'asc'])
-
-
-class _IBM_Boolean(sa_types.Boolean):
-
+RESERVED_WORDS = {'activate', 'disallow', 'locale', 'result', 'add', 'disconnect', 'localtime',
```
Comment (kadler) [inline sqlalchemy_ibmi/base.py]: Looks like the auto formatter hasn't done a great job here. We can move the code down and just leave an opening brace. Then all the rest of the lines can be un-indented to the same leve.

File: sqlalchemy_ibmi/base.py [hunk 4]
```diff
@@ -205,249 +245,245 @@ class XML(sa_types.Text):
 
 
 class DB2TypeCompiler(compiler.GenericTypeCompiler):
-
-
-    def visit_TIMESTAMP(self, type_):
+    """ Compiler for DB2 Types """
+    def visit_TIMESTAMP(self, type_, **_):
```
Comment (kadler) [inline sqlalchemy_ibmi/base.py]: What's with the trailing underscore after **?

File: sqlalchemy_ibmi/base.py [hunk 5]
```diff
@@ -205,249 +245,245 @@ class XML(sa_types.Text):
...
         def visit_true(self, expr, **kw):
             return '1'
-            
+
     def get_cte_preamble(self, recursive):
         return "WITH"
-        
-    def visit_now_func(self, fn, **kw):
+
+    def visitNowFunc(self):
```
Comment (kadler) [inline sqlalchemy_ibmi/base.py]: This doesn't fit in the Python naming convention. Why did it change?

File: sqlalchemy_ibmi/base.py [hunk 6]
```diff
@@ -676,88 +756,374 @@ class DB2Dialect(default.DefaultDialect):
...
+                       schema="QSYS2")
+
+    sys_table_constraints = Table(
+        "SYSCST", ischema, Column(
+            "CONSTRAINT_SCHEMA", CoerceUnicode, key="conschema"), Column(
+                "CONSTRAINT_NAME", CoerceUnicode, key="conname"), Column(
+                    "CONSTRAINT_TYPE", CoerceUnicode, key="contype"), Column(
+                        "TABLE_SCHEMA", CoerceUnicode, key="tabschema"), Column(
+                            "TABLE_NAME", CoerceUnicode, key="tabname"), Column(
+                                "TABLE_TYPE", CoerceUnicode, key="tabtype"), schema="QSYS2")
```
Comment (kadler) [inline sqlalchemy_ibmi/base.py]: Can we clean this up to look like the previous and next ones?

File: sqlalchemy_ibmi/base.py [hunk 7]
```diff
@@ -676,88 +756,374 @@ class DB2Dialect(default.DefaultDialect):
...
+        ischema,
+        Column(
+            "SEQUENCE_SCHEMA",
+            CoerceUnicode,
+            key="seqschema"),
+        Column(
+            "SEQUENCE_NAME",
+            CoerceUnicode,
+            key="seqname"),
+        schema="QSYS2")
```
Comment (kadler) [inline sqlalchemy_ibmi/base.py]: This is could also be formatted better, I think

File: sqlalchemy_ibmi/base.py [hunk 8]
```diff
@@ -676,88 +756,374 @@ class DB2Dialect(default.DefaultDialect):
...
-        
-    def normalize_name(self, name):
-        return self._reflector.normalize_name(name)
-
-    def denormalize_name(self, name):
-        return self._reflector.denormalize_name(name)
 
     def _get_default_schema_name(self, connection):
-        return self._reflector._get_default_schema_name(connection)
+        """Return: current setting of the schema attribute"""
```
Comment (kadler) [inline sqlalchemy_ibmi/base.py]: This looks totally different. I thought this PR was just linter changes.

File: sqlalchemy_ibmi/base.py (hunk 5 above)
Comment (naveen-ram) [reply sqlalchemy_ibmi/base.py]: This was an issue that pylint was complaining about.

File: sqlalchemy_ibmi/base.py (hunk 8 above)
Comment (naveen-ram) [reply sqlalchemy_ibmi/base.py]: I ran the linter after making the consolidation changes. Should I go through and lint on master even though those some of those files will be deleted in the consolidation PR.

Comment (kadler) [reply sqlalchemy_ibmi/base.py]: Yeah. It's a little more work, but the benefit is that by running the linter and formatter we can be relatively assured that the code being changed is just for linting, not any substantial changes.

I'd do it in two commits: one with the autoformatter run and one with manual adjustments. The autoformatter commit should contain the commands run to do the formatting as well.

Comment (kadler) [reply sqlalchemy_ibmi/base.py]: Let's also do enable full linting the PR in a separate PR.

**Comment by kadler** [comment]
This was made redundant by #18.

**Pull request closed without merging**